nested-spire/README.md

```diff
@@ -0,0 +1,226 @@
+       -spiffeID "spiffe://example.org/nestedA/workload" \
+       -selector "unix:uid:1001" \
+       -ttl 0
+
+   # Workload for nestedB deployment
+   docker-compose exec -T nestedB-server \
+       /opt/spire/bin/spire-server entry create \
+       -parentID "spiffe://example.org/spire/agent/x509pop/$(fingerprint nestedB/agent/agent.crt.pem)" \
+       -spiffeID "spiffe://example.org/nestedB/workload" \
+       -selector "unix:uid:1001" \
+       -ttl 0
+```
+
+The examples use the `$(fingerprint nestedA/agent/agent.crt.pem)` form again to show that the `-parentID` flag specifies the SPIFFE ID of the nested SPIRE Agent. The TTL flag value of zero indicates that the default value of 3600 seconds will be used. Finally, in both cases the unix selector assigns the SPIFFE ID to any process with a uid of 1001.
+
+Use the following Bash script to create the registration entries using the options just described:
+
+```console
+   bash scripts/create-workload-registration-entries.sh
+```
+
+## Run the Test
+
+Once both workload registration entries are propagated, let's test that SVIDs created in a nested configuration are valid in the entire trust domain, not only in the scope of the SPIRE Server that originated the SVID.
+
+The test consists of getting a JWT-SVID from the `nestedA-agent` SPIRE Agent and validating it using the `nestedB-agent`. In both cases, Docker Compose runs the processes using the uid 1001 to match the workload registration entries created in the previous section.
+
+Type this command to fetch the JWT-SVID on the `nestedA` SPIRE Agent and extract the token from the JWT-SVID:
+
+```console
+    token=$(docker-compose exec -u 1001 -T nestedA-agent \
+      /opt/spire/bin/spire-agent api fetch jwt -audience nested-test -socketPath /opt/spire/sockets/workload_api.sock | sed -n '2p')
+```
+
+Run the following command to validate the token from `nestedA` on the `nestedB` SPIRE Agent:
+
+```console
+    docker-compose exec -u 1001 -T nestedB-agent \
+        /opt/spire/bin/spire-agent api validate jwt -audience nested-test  -svid "${token}" \
+          -socketPath /opt/spire/sockets/workload_api.sock
+```
+
+The `nestedB` SPIRE Agent outputs the following:
+
+```console
+    SVID is valid.
+    SPIFFE ID : spiffe://example.org/nestedA/workload
+    Claims    : {"aud":["nested-test"],"exp":1595814190,"iat":1595813890,"sub":"spiffe://example.org/nestedA/workload"}
+```
+
+The output indicates that the JWT-SVID is valid. Also, although the SPIFFE ID is registered on `nestedA` rather than `nestedB`, the SPIFFE ID is still valid on the `nestedB` SPIRE Agent because the SPIRE Agents are in the same trust domain in the nested SPIRE topology.
+
+In SPIRE this is accomplished by propagating every JWT-SVID public signing key to the whole topology. In the case of X509-SVID, this is easily achieved because of the chaining semantics that X.509 has.
```

Comment:
can you add steps to validate x509 svids as well? basically you can use openssl to do that